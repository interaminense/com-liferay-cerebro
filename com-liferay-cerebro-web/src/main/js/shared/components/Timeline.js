import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

import Icon from './Icon';

class Timeline extends Component {
	render() {
		const {activeIndex, items} = this.props;

		return (
			<div class="timeline-root">
				{items.map(({href, title}, i) => {
					const active = activeIndex === i;
					const previousStep = activeIndex > i;

					const step = i + 1;

					return [
						<a
							class={getCN('step', {
								active,
								'previous-step': previousStep
							})}
							href={href}
							key={i}
						>
							<div class="title">{title}</div>

							<div class="circle">
								{!previousStep && step}

								{previousStep && <Icon symbol="check" />}
							</div>
						</a>,
						items.length !== i + 1 && (
							<span
								class={getCN('bar', {filled: previousStep})}
								key={`bar-${i}`}
							/>
						)
					];
				})}
			</div>
		);
	}
}

Timeline.PROPS = {
	activeIndex: Config.number(),
	items: Config.array()
};

export default Timeline;