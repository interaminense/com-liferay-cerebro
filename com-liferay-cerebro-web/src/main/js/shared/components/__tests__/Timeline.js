import Timeline from '../Timeline';

describe('Timeline', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Timeline({
			activeIndex: 1,
			items: [
				{
					title: 'This is a really long title for this step'
				},
				{title: 'Step 2'},
				{title: 'Step 3'},
				{title: 'Step 4'}
			]
		});

		expect(component).toMatchSnapshot();
	});
});