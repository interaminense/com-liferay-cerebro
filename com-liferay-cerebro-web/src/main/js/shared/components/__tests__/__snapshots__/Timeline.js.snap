// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should render 1`] = `
<div class="timeline-root">
  <a class="step previous-step">
    <div class="title">This is a really long title for this step</div>
    <div class="circle">
      <svg class="icon-root lexicon-icon lexicon-icon-check">
        <use xlink:href="/images/lexicon/icons.svg#check"></use>
      </svg>
    </div>
  </a>
  <span class="bar filled"></span>
  <a class="step active">
    <div class="title">Step 2</div>
    <div class="circle">2</div>
  </a>
  <span class="bar"></span>
  <a class="step">
    <div class="title">Step 3</div>
    <div class="circle">3</div>
  </a>
  <span class="bar"></span>
  <a class="step">
    <div class="title">Step 4</div>
    <div class="circle">4</div>
  </a>
</div>
`;
